feature(Monthly Balance Analysis)

lomamech opened the pull request:
- Add new report for Analysis Monthly Balance of accounts

closes #3803

lomamech committed:
feature(Monthly Balance Analysis)
- Add new report for Analysis Monthly Balance of accounts

closes #3803

lomamech committed:
Sanitaze code

lomamech committed:
Sanitaze code

mbayopanda commented on client/src/i18n/en/report.json:
```diff
@@ -69,6 +69,10 @@
     "INCOME_REPORT": "Income Report",
     "MONTHLY_BALANCE": "Monthly Balance",
+    "MONTHLY_ACCOUNT_ANALYSIS": {
+      "DESCRIPTION": "This report allows for monthly analysis of accounts",
```
Could you be more explicit, which kind of analysis ?

mbayopanda commented on client/src/i18n/en/report.json:
```diff
@@ -69,6 +69,10 @@
     "MONTHLY_BALANCE": "Monthly Balance",
+    "MONTHLY_ACCOUNT_ANALYSIS": {
+      "DESCRIPTION": "This report allows for monthly analysis of accounts",
+      "TITLE": "Monthly Balance analysis"
```
`Monthly Balance Analysis` is not explicit for what for the report is.

mbayopanda replied:
Thank you for the work, i saw the report and i think a good name can be something like "**Monthly Balance**" or "**Periodic Balance**" because it shows up the balance of a given period.

mbayopanda commented on client/src/i18n/fr/report.json:
```diff
@@ -63,6 +63,10 @@
     "MONTHLY_BALANCE" : "Balance Mensuelle",
+    "MONTHLY_ACCOUNT_ANALYSIS": {
+      "DESCRIPTION": "Ce rapport permet de faire l'analyse mensuel des comptes",
+      "TITLE": "Analyse mensuel de la Balance"
```
Which kind of analysis ?

mbayopanda replied:
"Balance par mois" or "Balance par periode" can be good as title, since the report is just the balance for a given period

mbayopanda commented on client/src/i18n/fr/tree.json:
```diff
@@ -67,6 +67,7 @@
     "IPR_TAX_CONFIGURATION": "Configuration de la taxe IPR",
     "JOURNAL_VOUCHER":"Ajout Transactions",
     "LOCATION":"Localisations",
+    "MONTHLT_ACCOUNT_ANALYSIS": "Analyse mensuel de la Balance",
```
`MONTHLY_ACCOUNT_ANALYSIS` may be

mbayopanda commented on client/src/modules/reports/generate/monthlyBalanceAnalysis/monthlyBalanceAnalysis.html:
```diff
@@ -0,0 +1,103 @@
+              period-id="ReportConfigCtrl.reportDetails.period_id"
+              on-select-callback="ReportConfigCtrl.onSelectPeriod(period)">
+            </bh-period-selection>            
+            <hr>
```
Could you remove these `<hr>` the final result let a lot of space in the user interface.

![image](https://user-images.githubusercontent.com/5445251/61890980-979cb080-af00-11e9-8d10-9c56e0444122.png)

mbayopanda commented on client/src/modules/reports/generate/monthlyBalanceAnalysis/monthlyBalanceAnalysis.html:
```diff
@@ -0,0 +1,103 @@
+              on-select-callback="ReportConfigCtrl.onSelectPeriod(period)">
+            </bh-period-selection>            
+            <hr>
+            <div class="panel-body" ng-class="{'has-error' : ConfigForm.$submitted && ConfigForm.allAccount.$invalid }">
```
Is it possible to remove this `css` class `panel-body` here

mbayopanda commented on client/src/modules/reports/generate/monthlyBalanceAnalysis/monthlyBalanceAnalysis.html:
```diff
@@ -0,0 +1,103 @@
+                </div>
+              </div>
+            </div>
+            <hr>
```
Could you remove this `<hr>`

mbayopanda commented on server/controllers/finance/reports/monthlyBalance/index.js:
```diff
@@ -0,0 +1,176 @@
+      filterByAccount = selectAccountParent(accountNumber);
+    } else {
+      filterByAccount = ``;
+    }
```
You can write : 
```js
let filterByAccount = accountNumber ? selectAccountParent(accountNumber) : '';
```

mbayopanda commented on server/controllers/finance/reports/monthlyBalance/index.js:
```diff
@@ -0,0 +1,176 @@
+    if (accountNumber) {
+      filterByAccount = selectAccountParent(accountNumber);
+    } else {
+      filterByAccount = ``;
```
For your information, back quotes ```` are used for multi lines or if we want to insert a variable or an expression : 
```js
`My text and the value of the variable is ${myVariable}` 
```
or in multi-lines : 
```js
`
Line 1
Line 2
etc.
`
```
In this case, you can just write `filterByAccount = ''`;

mbayopanda reviewed (commented):
A made a first review

lomamech committed:
improvement(MonthlyBalance)
- Sanitaze code and improve report and translate key

mbayopanda commented on client/src/i18n/en/tree.json:
```diff
@@ -67,6 +67,7 @@
     "IPR_TAX_CONFIGURATION" : "IPR Tax Configuration",
     "JOURNAL_VOUCHER" : "Journal Voucher",
     "LOCATION" : "Location Management",
+    "MONTHLY_ACCOUNT_ANALYSIS": "Monthly Balance analysis",
```
Since it is just a balance of a given period, i think it is no more necessary to add `Analysis`

mbayopanda commented on client/src/i18n/fr/tree.json:
```diff
@@ -67,6 +67,7 @@
     "IPR_TAX_CONFIGURATION": "Configuration de la taxe IPR",
     "JOURNAL_VOUCHER":"Ajout Transactions",
     "LOCATION":"Localisations",
+    "MONTHLY_ACCOUNT_ANALYSIS": "Analyse mensuel de la Balance",
```
"Balance par mois"

lomamech replied:
Balance Mensuel  would be better than Monthly Balance

mbayopanda commented on server/controllers/finance/reports/monthlyBalance/report.handlebars:
```diff
@@ -0,0 +1,64 @@
+      <h3 class="text-center text-uppercase">
+        <strong>{{translate 'REPORT.MONTHLY_ACCOUNT_ANALYSIS.TITLE'}}</strong> <br>
+      </h3>
+      <hr>
```
Could you remove this `<hr>`

mbayopanda commented on server/controllers/finance/reports/monthlyBalance/report.handlebars:
```diff
@@ -0,0 +1,64 @@
+
+      <h5 style="margin:15px; font-weight:bold" class="text-center text-uppercase">
+        {{periodLabel}}
+      </h5>
```
Since we select the period before accounts, i think `period` must be above `account`.

Here is a proposal: 
```html
<h3 class="text-center text-bold text-uppercase">{{translate 'REPORT.MONTHLY_ACCOUNT_ANALYSIS.TITLE'}}</h3>
<h3 class="text-center text-uppercase">{{periodLabel}}</h3>
<h4 class="text-center">{{this.accountNumber}}: {{this.accountLabel}}</h4>
```

mbayopanda reviewed (commented):
Good job @lomamech , i made just some comments the work is well done 👍

lomamech committed:
improve(Monthly Balance)
- Rename Monthly Balance instead Monthly Balance Analysis
- Sanitaze code and refactor Report Design

mbayopanda reviewed (approved):
bors r+
